Include FETCH responses in priority scheduling algorithm (#816)
moq-wg/moq-transport: draft-ietf-moq-transport.md

MoQT maintains priorities between different _schedulable objects_.
A schedulable object in MoQT is either:

1. An object in response to a SUBSCRIBE that belongs to a subgroup where

>    that object would be the next object to be sent in that subgroup.

Reviewer: ```suggestion
   that object is the next object in that subgroup.
```

> 1. An object in response to a SUBSCRIBE that belongs to a track with
>    delivery preference Datagram.
> 1. An object in response to a FETCH where that object would be the next
>    object in the response.

Reviewer: ```suggestion
2. An object in response to a SUBSCRIBE that belongs to a track with
   delivery preference Datagram.
3. An object in response to a FETCH where that object is the next
   object in the Track.
```

Author: Track is ambiguous for FETCH, since there can be > 1 FETCH for the same track at the same time.

   the one with **the highest subscriber priority** is sent first.
1. If two objects have the same subscriber priority, but a different publisher
   priority, the one with **the highest publisher priority** is sent first.

> 1. If two objects in response to the same request have the same subscriber

Reviewer: ```suggestion
2. If two objects in response to the same request have the same subscriber
```

   and publisher priority, but belong to two different groups of the same track,
   **the group order** of the associated subscription is used to
   decide the one that is sent first.

> 1. If two objects in response to the same request belong to the same group of

Reviewer: ```suggestion
3. If two objects in response to the same request belong to the same group of
```

   with delivery preference Subgroup), or **the lowest Object ID** (for tracks
   with delivery preference Datagram) is sent first.

When Objects are delivered via FETCH, only the subscriber priority is
considered. If a FETCH response has the same subscriber priority as

> another schedulable Object, the order is implementation specific.

Reviewer 2: Do we want to make the publisher priority optional in the subgroup header, since it's irrelevant for fetches?

Author: I think priority is well baked into subgroups - though if we get deep into bit packing the data plane, I can imagine trying to get clever.

> When objects are delivered via FETCH, only the subscriber priority is
> considered.  If a FETCH response stream has the same subscriber priority as
> another schedulable object, the order is implementation defined.

Reviewer: ```suggestion
When Objects are delivered via FETCH, only the subscriber priority is
considered. If a FETCH response has the same subscriber priority as
another schedulable Object, the transmission order is implementation
specific.
```

Author: Happy to change this, though the next paragraph has this:

> This algorithm does not provide a well-defined ordering for objects that belong to different subscriptions, but have the same subscriber and publisher priority.  The ordering in those cases is implementation-defined,

Author: @ianswett do you want me to take your suggestion in light of the subsequent paragraph?

Reviewer: I like the wording "implementation specific" better than implementation defined, but it's a nit for sure.

My suggestion of adding "transmission" is probably not helpful given how the rest is worded, so I'll remove that change.

   the one with **the highest subscriber priority** is sent first.
1. If two objects have the same subscriber priority, but a different publisher
   priority, the one with **the highest publisher priority** is sent first.
2. If two objects in response to the same request have the same subscriber
   and publisher priority, but belong to two different groups of the same track,
   **the group order** of the associated subscription is used to
   decide the one that is sent first.
3. If two objects in response to the same request belong to the same group of

>    the same track, the one with **the lowest Subgroup ID** (for tracks

Author: I recognize those two overlap but I don't think it reads too awkwardly so I'm going to leave it
